File: data_structures_and_algorithms/Code_401/class-04/solutions/interview-01/python/sum_of_rows/sum_of_rows.py
def sum_rows(matrix):
    """
    For input matrix of n rows
    return list of n length with the sums of each row
    """

    sums = []

    for row in matrix:

        sum = 0

        for column in row:

            # handle None
            column_val = column or 0

            sum += column_val

        sums.append(sum)

    return sums

File: data_structures_and_algorithms/Code_401/class-04/solutions/interview-01/javascript/sumOfRows/sumOfRows.js
'use strict';

function sumOfRows(matrix) {

  let sums = [];

  for (let row of matrix) {
    let sum  = 0;

    for (let value of row) {
      sum += value;
    }
    sums.push(sum);
  }

  return sums;
}

module.exports = sumOfRows;
